inference-engine/src/transformations/src/transformations/common_optimizations/common_optimizations.cpp

```diff
@@ -89,6 +90,11 @@ bool ngraph::pass::CommonOptimizations::run_on_function(std::shared_ptr<ngraph::
     // This pass must be called first in pipeline
     manager.register_pass<ngraph::pass::InitNodeInfo>();
     manager.register_pass<ngraph::pass::SimplifyShapeOfSubGraph>();
+
+    // This transformation must be called before constant folding as it folds
+    // ShapeOf node, which is required to be checked in dropout subgraph
+    manager.register_pass<ngraph::pass::DropoutWithRandomUniformReplacer>();
```

Comment:
Do we have a model in pre-commit for which this transformation is applied? As I understand the transformation is sensitive to reordering of other transformations. Can we track this case if `Mul-Add` fusion transformation or constant folding change their order so that your transformation stops working?

Reply:
Yes, we have a model in pre-commit for which this transformation is applied. Without this transformation e2e fails as reference implementation is not merged yet. 
I also checked all benchmark models for which original MO transformation was applied and checked this transformation on them.

Reply:
Yes, this transformation is sensitive to constant folding. So I added comment that it should be run before constant folding.

Reply:
After removing of ShapeOf from pattern transformation is no longer Constant Folding dependent.